File: c/sorting/bubble_sort.c
#include<stdio.h>


int bubblesort(int a[],int n){
	int i,j;
	for(i=0;i<n-1;i++){
		int swapped = 0;
		for(j=0;j<n-1-i;j++){
			if(a[j]>a[j+1]){
				int temp = a[j];	//Swaping
				a[j] = a[j+1];
				a[j+1] = temp;
				swapped = 1;
			}
		}
		if(swapped == 0){
			return 0;
		}
	}
}

int main(){
	printf("Enter the number of Elements\n");
	int n,i;
	scanf("%d",&n);
	int a[n];
	for(i=0;i<n;i++){
		scanf("%d",&a[i]);
	}
	bubblesort(a,n);
	printf("Sorted array is :\n");
	for(i=0;i<n;i++){
		printf("%d ",a[i]);
		printf("\n");
	}
}

File: sorting/bubble_sort/bubbleSort.java
public class bubbleSort {

    public static void doBubbleSort(int a[]){
        int n = a.length;
        for (int i = 0; i < n - 1; i++) {
            for (int j = 0; j < n-i-1; j++) {
                if(a[j+1]<a[j]){
                    int temp = a[j+1];
                    a[j+1] = a[j];
                    a[j] = temp;
                }
            }
        }
    }

    public static void main(String[] args) {
        int a[] = {10, 23, 0, 50, 12, 99, 56, 2, 1, 11};
        System.out.println("Unsorted Array: ");
        for (int i = 0; i < a.length; i++) {
            System.out.print(a[i] + " ");
        }
        System.out.println();
        doBubbleSort(a);
        System.out.println("Sorted array using bubble sort is:");
        for (int i = 0; i < a.length; i++) {
            System.out.print(a[i] + " ");
        }
    }
}
